src/OpenTelemetry.Exporter.Prometheus.HttpListener/Internal/PrometheusSerializer.cs

```diff
@@ -222,10 +222,20 @@ public static int WriteLabel(byte[] buffer, int cursor, string labelKey, object
         buffer[cursor++] = unchecked((byte)'"');
 
         // In Prometheus, a label with an empty label value is considered equivalent to a label that does not exist.
-        cursor = WriteLabelValue(buffer, cursor, labelValue?.ToString() ?? string.Empty);
+        cursor = WriteLabelValue(buffer, cursor, GetLabelValueString(labelValue));
         buffer[cursor++] = unchecked((byte)'"');
 
         return cursor;
+
+        static string GetLabelValueString(object labelValue)
```

Comment:
Consider adding a TODO comment here pointing to https://github.com/open-telemetry/opentelemetry-dotnet/issues/4822#issuecomment-1707328495.

Reply:
Done